app/javascript/packs/BraintreeForm.js

```diff
@@ -0,0 +1,35 @@
+import {
+  BraintreeFormWrapper,
+  createBraintreeClient
+} from 'PaymentGateways'
+import client from 'braintree-web/client'
+import { safeFromJsonString } from 'utilities/json-utils'
+
+const CONTAINER_ID = 'braintree-form-wrapper'
+
+document.addEventListener('DOMContentLoaded', async () => {
+  const container = document.getElementById(CONTAINER_ID)
+  if (!container) {
+    return
+  }
+
+  const { clientToken, billingAddress, threeDSecureEnabled, formActionPath, countriesList, selectedCountryCode } = container.dataset
+  const billingAddressParsed = safeFromJsonString(billingAddress)
+  for (let key in billingAddressParsed) {
+    if (!billingAddressParsed[key]) {
+      billingAddressParsed[key] = ''
+    }
+  }
+  const braintreeClient = await createBraintreeClient(client, clientToken)
```

Comment:
Do we know if this is supposed to be immediate? If not, the whole page could freeze (imagine Braintree server is overloaded and takes 2 min to respond).

This should be instantiated within a Component with a `isLoading` state that shows/hides a loading spinner and prevents any user interaction without blocking the page. We could do this in a follow-up but let's create a TODO or a JIRA.

Reply:
OK, I'll create the Jira